File: Game.Application/GameJS/src/terrain-generation/terrain-specifications/TerrainSituation.ts

import { HumidityCategory } from "../HumidityCategory";
import { TemperatureCategory } from "../TemperatureCategory";
import { BiomeCategory } from "../BiomeCategory";
import { AltitudeCategory } from "../WaterCategory";

export interface TerrainSituation {
    readonly altitude: number;
    readonly heat: number;
    readonly humidity: number;
    readonly feature: number;
    readonly temperatureCategory: TemperatureCategory;
    readonly humidityCategory: HumidityCategory;
    readonly biomeCategory: BiomeCategory;
    readonly altitudeCategory: AltitudeCategory;
  }

File: Game.Domain/Terrain/TerrainSituation.cs
namespace Game.Domain.Terrain
{
    public class TerrainSituation
    {
        public float Altitude { get; set; }
        public float Heat { get; set; }
        public float Humidity { get; set; }
        public float Feature { get; set; }
        public TemperatureCategory TemperatureCategory { get; set; } 
        public HumidityCategory HumidityCategory { get; set; } 
        public BiomeCategory BiomeCategory { get; set; } 
        public AltitudeCategory AltitudeCategory { get; set; } 
    }
}